File: src/test/java/spoon/test/literal/testclasses/BasedLiteral.java
package spoon.test.literal.testclasses;

public class BasedLiteral {

    int i1 = 42;
    int i2 = 0;
    int i3 = 00;
    int i4 = 042;
    int i5 = 0x42;
    int i6 = 0X42;
    int i7 = 0_1_4_2;
    int i8 = 0b1;
    int i9 = 0b1010;

    long l1 = 42L;
    long l2 = 0L;
    long l3 = 00L;
    long l4 = 042L;
    long l5 = 0x42L;
    long l6 = 0X42L;
    long l7 = 0b0L;
    long l8 = 0b1010L;

    float f1 = 42.42f;
    float f2 = 0042.0f;
    float f3 = 00f;
    float f4 = 0f;
    float f5 = 0.0f;
    float f6 = 0x1.2p7f;
    float f7 = 0X1.2p7f;

    double d1 = 0.0;
    double d2 = 0.;
    double d3 = 042.;

    char c1 = 'c';
    String s1 = "hello";
}

File: spoon-kotlin/src/test/kotlin/spoon/test/literal/testclasses/BasedLiteral.kt
package spoon.test.literal.testclasses

class BasedLiteral {
    val i1 = 42
    val i2 = 0
    val i3 = 0x42
    val i4 = 0X43
    val i5 = 1_4_2
    val i6 = 0b1
    val i7 = 0b1010

    val l1: Long = 42
    val l2 = 42L
    val l3 = 0x42L
    val l4 = 0X43L
    val l5: Long = 0xa
    val l6: Long = 0b01
    val l7 = 0b1010L

    val f1: Float = 42.42F
    val f2 = 42.43f
    val f3 = .4f

    val d1 = 0.6
    val d2 = .3

    val c1 = 'c'

    val s1 = "hello"

    val b1: Byte = 1
    val b2: Byte = 0x2
    val b3: Byte = 0b11
    val b4: Byte = 0x1f

    val sh1: Short = 11
    val sh2: Short = 0xc
    val sh3: Short = 0b1101
}